.github/workflows/build-and-test.yml

```diff
@@ -21,7 +21,7 @@ jobs:
         uses: actions/cache@v2
         with:
           path: node_modules
-          key: ${{ runner.os }}-node-${{ hashFiles('package-lock.json') }}
+          key: ${{ runner.os }}-node-modules-${{ hashFiles('package-lock.json') }}
```

Comment:
this needs to be updated in the publish job, too.

Reply:
I saw that and I think we're doing something different in publish job

1. `restoreKeys` contain `package.json` instead of `package-lock.json` in the key value and I think it would always be different right?
2. we're not checking `cache-hit` and every time we're doing `npm ci` so cache part seems unused here.

Reply:
```
name: Publish artifacts
on:
  # Will only run when release is published.
  release:
    types:
      - created
  workflow_dispatch:

jobs:
  publish-docker:
    runs-on: ubuntu-20.04
    steps:
      # Set fetch-depth: 0 to fetch commit history and tags for use in version calculation
      - name: Check out code
        uses: actions/checkout@v2.3.4
        with:
          fetch-depth: 0

      - name: Cache node modules
        uses: actions/cache@v2
        with:
          path: ~/.npm
          key: ${{ runner.os }}-node-${{ hashFiles('package.json') }}-${{ hashFiles('package-lock.json') }}
          restore-keys: |
            ${{ runner.os }}-node-${{ hashFiles('package.json') }}-
            ${{ runner.os }}-node-

      - name: NPM Install
        run: npm ci

      - name: Build
        run: npm run build:ci
```

Reply:
whoops - yeah, that one is different but shouldn't be. It will still work, just off fresh node modules each time - so we can come back to that as tech debt.

Reply:
should we go ahead and merge this PR?